pylint/checkers/typecheck.py

```diff
@@ -579,6 +581,40 @@ def _emit_no_member(
     return True
 
 
+def _enum_has_attribute(owner, node) -> bool:
+    enum_def = owner
+    # Traverse the AST to find the Enum ClassDef
+    while not isinstance(enum_def, astroid.ClassDef):
+        enum_def = enum_def.parent
+
+    # Find __new__
+    dunder_new = next((m for m in enum_def.methods() if m.name == "__new__"), None)
+    if dunder_new is None:
+        return False
+
+    # Get the returned object
+    returned_obj_name = next(
+        (c.value for c in dunder_new.get_children() if isinstance(c, astroid.Return)),
+        None,
+    )
+    if returned_obj_name is None:
+        return False
+
+    enum_attributes = set()
+    # Iterate over children and find Enum attr assignments
+    for child in dunder_new.get_children():
+        if not isinstance(child, astroid.Assign):
+            continue
```

Comment:
```suggestion
    for child in dunder_new.nodes_of_class(astroid.Assign):
```
Maybe something to consider ? This is not equivalent as it's also searching in children of children.

Reply:
Thanks, didn't know this method existed!
I think it _is_ better to check the children as it would catch members defined anywhere in the function def tree.
If someone were to define a method within `__init__` or `__new__` then the checker could do the wrong thing. However, my response would be "don't do that" as it's kinda nasty.